### DISCUSSION TITLE
Bulk processing/deletion with doctrin via Command - out of memory

### DISCUSSION THREAD

**Discussion by UtechtDustin** [symfony/symfony#54401] [Q&A]
We tried to bulk delete an entity which is older than 24 hours (~2200 Entities).   
We followed the [documentation of doctrine](https://www.doctrine-project.org/projects/doctrine-orm/en/3.1/reference/batch-processing.html#iterating-results) and we always run out of memory.    

After some time/debugging we found out that the sql logger seems to be the problem.   
We found the parameter `--no-debug` in an other issue(https://github.com/doctrine/dbal/issues/5815) which fixed the problem.  
But to add an extra parameter which maybe will be forgotten in the future, looks odd for me.
So i found a way to do it via code
```
$this->entityManager->getConnection()->getConfiguration()->setSQLLogger();
```
which is depcreated, so i also tested this:
```
$this->entityManager->getConnection()->getConfiguration()->setMiddlewares([new Middleware(new NullLogger())]);
```

Both don't helped out, we still run out of memory.  
What is the common/best way to do batch processing/deletion in a command ?

**Comment by MatTheCat** [comment]
Are you sure you cannot use a [DQL DELETE](https://www.doctrine-project.org/projects/doctrine-orm/en/3.1/reference/batch-processing.html#dql-delete)? This would be the best solution.

Else, adding the `--no-debug` flag would be the simplest: changing the SQL logger at runtime is not possible in your context, so you’d need to proxy it like https://github.com/symfony/symfony/issues/46158#issuecomment-1163363618 shows.

**Reply by UtechtDustin** [reply]
I can't use DQL because i have to use a join, which is not supported as i read.

**Reply by alexandre-le-borgne** [reply]
The --no-debug flag has a very significant impact and avoids major doctrine memory leaks

To reduce memory usage, you can also add at the end of each loop:

            gc_enable();
            gc_collect_cycles();
